### DISCUSSION TITLE
Regex in zos_lineinfile

### DISCUSSION THREAD

**Discussion by ghost** [ansible-collections/ibm_zos_core#907] [Q&A] [answered]
Hi guys!

I have the following entry in a PDS member:

//* 06/24/2021 D160882 - Creation V560

I'm using the zos_lineinfile module to find and delete the string D160882.

When I put exactly that string in the regex parameter, it finds and deletes the line as requested:

   zos_lineinfile:
     src: "{{ hlq_amb }}.PLEX{{ hlq_lib }}.PROCLIB.ANS({{ cic_new }})"
     regexp: '.*D160882.*'
     state: absent
   register: result_target

However, when I actually try to use a regular expression to find a string that starts with a letter and is followed by 6 numbers, it doesn't.
I've tried several different expressions, such as: '^(.*)[A-Z][0-9]{6}(.*)$)', '\b[A-Za-z][0-9]{6,7}\b', '\\b[A-Za-z][0-9]{6,7}\\b' but none of them find it.

   zos_lineinfile:
     src: "{{ hlq_amb }}.PLEX{{ hlq_lib }}.PROCLIB.ANS({{ cic_new }})"
     regexp: '.*[A-Z][0-9]{6}.*'
     state: absent
   register: result_target

This is the message when it does not find:

![image](https://github.com/ansible-collections/ibm_zos_core/assets/105386204/2e7a5fbd-0268-4e9c-ae77-2affecbe70a6)

Could someone help me please?

**Comment by ddimatos** [comment]
Hello @fabiobncbb , which version of the ibm_zos_core collection are you using and which version of zoau so I may try to reproduce.

**Reply by ghost** [reply]
Hello @ddimatos!

I'm using zos_core 1.6 and tried it in two different environments that i have (lab and desenv). One is with ZOAU 1.1 and other is with ZOAU 1.2

**Comment by ddimatos** [comment, marked as answer]
I tried this on the supported ZOAU 1.2.2 and ibm_zos_core version 1.6.0 and found that there is a bug in our dependency, thus I have opened a bug with them and noted this is a community reported issue #913 , please follow the issue `913` for resolution. I am provide some work around below, I am happy to discuss them further but for now it seems we can consider this mitigated and close the issue. 
 
The are some work arounds I can think of:
1) Use `zos_copy` and copy the content to USS, tag it (*chtag`) as UTF-8 and then use the community `blockinfile` or `shell` module to run `sed` or `awk` commands from USS, then `zos_copy` the data back to the data set. 
2) Use `zos_encode` to encode the data set to USS as UTF8, then use the community `shell` or blockfinfile module. 
3) Use `zos_fetch` , then perform operations locally on the controller and then with `zos_copy`, copy it back to MVS. 

I will create an issue tomorrow and report it back here, below is my playbook recreate. 
 
My recreate is:
```
    - name: Create seq data set
      zos_data_set:
        name: ANSIBLE.LINEFILE.SEQ
        type: SEQ
        space_primary: 3
        space_type: M
        record_format: FB
        record_length: 1024
        replace: yes
      register: result

    - debug:
       var: result

    - name: Populate data set
      zos_copy:
        content: |1
          D160882
          D160882 with more info
          ED160882
          This is a line
        dest: ANSIBLE.LINEFILE.SEQ
        force: yes
      register: result

    - debug:
       var: result

    - name: Update Data set by deleting a line
      zos_lineinfile:
        src: ANSIBLE.LINEFILE.SEQ
        regexp: '[A-Za-z][0-9]{6}'
        state: absent
      register: result

    - name: Response
      debug:
        msg: "{{ result }}"

    - name: Cat the data set
      command: "dcat 'ANSIBLE.LINEFILE.SEQ'"
      register: result

    - name: Response
      debug:
        msg: "{{ result }}"
```
